src/EventSourcingDb/Client.cs

```diff
@@ -152,8 +152,9 @@ public async Task<IReadOnlyList<Event>> WriteEventsAsync(
             using var response = await _httpClient.SendAsync(request, token).ConfigureAwait(false);
             if (response.StatusCode != HttpStatusCode.OK)
             {
+                var errorResponse = await response.Content.ReadAsStringAsync(token).ConfigureAwait(false);
                 throw new HttpRequestException(
-                    message: "Unexpected status code.", inner: null, statusCode: response.StatusCode
+                    message: $"Unexpected status code - {errorResponse}", inner: null, statusCode: response.StatusCode
```

Comment:
Basically, I'm fine with this, I'd just suggest a slightly different formatting, so that things better align with our typical style:


```suggestion
                    message: $"Unexpected status code ('{errorResponse}').", inner: null, statusCode: response.StatusCode
```